modules/web_application/main.tf

```diff
@@ -65,6 +68,162 @@ resource "dynatrace_web_application" "web_application" {
     ignore_case                  = true
     query_parameter_cleanups     = ["cfid", "phpsessid", "__sid", "cftoken", "sid"]
     split_user_actions_by_domain = true
+    xhr_action_naming_rules {
+      rule {
+        template = "{userInteraction (default)} on {elementIdentifier (default)} in {pageUrlPath (default)}"
+      }
+    }
+  }
+  meta_data_capture_settings {
+    capture {
+      capturing_name  = "document.referrer"
+      name            = "Referrer"
+      public_metadata = false
+      type            = "JAVA_SCRIPT_VARIABLE"
+      unique_id       = 1
+      use_last_value  = false
+    }
+    capture {
+      capturing_name  = "navigator.userAgent"
+      name            = "User agent"
+      public_metadata = false
+      type            = "JAVA_SCRIPT_VARIABLE"
+      unique_id       = 2
+      use_last_value  = false
+    }
+    capture {
+      capturing_name  = "utm_source"
+      name            = "UTM source"
+      public_metadata = false
+      type            = "QUERY_STRING"
+      unique_id       = 3
+      use_last_value  = false
+    }
+    capture {
+      capturing_name  = "utm_medium"
+      name            = "UTM medium"
+      public_metadata = false
+      type            = "QUERY_STRING"
+      unique_id       = 4
+      use_last_value  = false
+    }
+    capture {
+      capturing_name  = "utm_campaign"
+      name            = "UTM campaign"
+      public_metadata = false
+      type            = "QUERY_STRING"
+      unique_id       = 5
+      use_last_value  = false
+    }
+    capture {
+      capturing_name  = "utm_term"
+      name            = "UTM term"
+      public_metadata = false
+      type            = "QUERY_STRING"
+      unique_id       = 6
+      use_last_value  = false
+    }
+    capture {
+      capturing_name  = "utm_content"
+      name            = "UTM content"
+      public_metadata = false
+      type            = "QUERY_STRING"
+      unique_id       = 7
+      use_last_value  = false
+    }
+  }
+  user_action_and_session_properties {
+    property {
+      aggregation                   = "FIRST"
+      display_name                  = "Referrer"
+      id                            = 2
```

Comment:
Should the first 'id' be '1'?

Reply:
Good spot! I'd copied them from a plan I'd done, but they must have created 8 originally and then removed one so it started from 2!
Corrected now.